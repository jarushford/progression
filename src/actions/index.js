export const getTrainingData = (data) => ({
  type: 'GET_TRAINING',
  data
})

export const getProjects = (data) => ({
  type: 'GET_PROJECTS',
  data
})

export const clearProjects = () => ({
  type: 'CLEAR_PROJECTS'
})

export const getAscents = (data) => ({
  type: 'GET_ASCENTS',
  data
})

export const clearAscents = () => ({
  type: 'CLEAR_ASCENTS'
})

export const toggleDiscipline = (discipline) => ({
  type: 'TOGGLE_DISCIPLINE',
  discipline
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
})

export const toggleComplete = (key) => ({
  type: 'TOGGLE_COMPLETE',
  key
})
