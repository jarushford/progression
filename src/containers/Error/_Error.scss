.error {
    margin: 150px auto;
    text-align: center;
    background: $darkGrey;
    padding: 20px;
    color: $accent;

    button {
      @include button();
      margin: 30px;
    }
}