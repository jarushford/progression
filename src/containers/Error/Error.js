import React from 'react'
import { connect } from 'react-redux'
import { clearError } from '../../actions'
import '../../main.scss'

function Error({ error, clearError }) {
  return (
    <section className="error">
      <h2>{error}</h2>
      <h2>Something seems to have gone wrong!</h2>
      <button onClick={clearError}>Return to site</button>
    </section>
  )
}

const mapStateToProps = (state) => ({
  error: state.error
})

const mapDispatchToProps = (dispatch) => ({
  clearError: () => dispatch(clearError())
})

export default connect(mapStateToProps, mapDispatchToProps)(Error)