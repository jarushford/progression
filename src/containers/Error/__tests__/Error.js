import React from 'react'
import { shallow } from 'enzyme'
import { clearError } from '../../../actions'
import { mapStateToProps, mapDispatchToProps, Error } from '../Error'

describe('Error', () => {
  let mockError

  beforeEach(() => {
    mockError = 'Oh no!'
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<Error error={mockError} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should mapStateToProps with data from store', () => {
    const mockState = {
      error: 'Oh no!'
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(mockState)
  })

  it('should mapDispatchToProps with a method clearError', () => {
    const mockDispatch = jest.fn()
    const expected = clearError()

    const mappedProps = mapDispatchToProps(mockDispatch)

    mappedProps.clearError()

    expect(mockDispatch).toBeCalledWith(expected)
  })
})