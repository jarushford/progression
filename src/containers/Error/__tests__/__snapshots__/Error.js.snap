// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error
    error="Oh no!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Oh no!
        </h2>,
        <h2>
          Something seems to have gone wrong!
        </h2>,
        <button>
          Return to site
        </button>,
      ],
      "className": "error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oh no!",
        },
        "ref": null,
        "rendered": "Oh no!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something seems to have gone wrong!",
        },
        "ref": null,
        "rendered": "Something seems to have gone wrong!",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return to site",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Return to site",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Oh no!
          </h2>,
          <h2>
            Something seems to have gone wrong!
          </h2>,
          <button>
            Return to site
          </button>,
        ],
        "className": "error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oh no!",
          },
          "ref": null,
          "rendered": "Oh no!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something seems to have gone wrong!",
          },
          "ref": null,
          "rendered": "Something seems to have gone wrong!",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return to site",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Return to site",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
